var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -206,6 +213,11 @@ def configure_args(self):
         if hdf5_hl.satisfies("~shared"):
             libs.append(hdf5_hl.libs.link_flags)
 
+        if "~shared" in self.spec:
```

Comment:
These should come in via `hdf5_hl.libs.link_flags`? Same as for `szlib`, `zstd`, ...